library/ViewPager.js

```diff
@@ -178,6 +178,19 @@ export default class ViewPager extends Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.initialPage !== this.props.initialPage) {
```

Comment:
if initialPage changed , then scroll to nextProps.initialPage.
this code should be put in componentWillReceiveProps not in componentDidUpdate